src/widgets/PieChart.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2018
+ */
+
+import React, { useState, useEffect } from 'react';
+import { View, StyleSheet } from 'react-native';
+import PropTypes from 'prop-types';
+import { VictoryLabel, VictoryPie } from 'victory-native';
+import { APP_FONT_FAMILY, GREY } from '../globalStyles';
+
+/**
+ * A charting widget for displaying Pie chart report.
+ *
+ * @prop  {string}       id      The report ID.
+ * @prop  {string}       title   The title of the report.
+ * @prop  {string}       type    The type of chart to use to display the report,
+ *                               options are BarChart, LineChart and PieChart.
+ * @prop  {data}         array   An array of {x, y} datapoints to plot.
+ * @prop  {width}        number  The width of the parent container.
+ * @prop  {height}       number  The height of the parent continer.
+ */
+
+export const ReportChart = ({ report: initialReport }, props) => {
+  const [report, setReport] = useState(initialReport);
+  const [width, setWidth] = useState(null);
+  const [height, setHeight] = useState(null);
+
+  // Same affect as componentDidUpdate
+  useEffect(() => {
+    if (report.id !== props.report.id) setReport(props.report);
+  });
+
+  // Victory Native sizes are set using absolute values. Parents dimensions are used to
+  // calculate relative values for width and height for each chart.
+  const onLayout = event => {
+    setWidth(event.nativeEvent.layout.width);
+    setHeight(event.nativeEvent.layout.height);
+  };
+
+  const renderPieChart = () => {
+    const {
+      padVertical,
+      padHorizontal,
+      innerRadius,
+      labelRadius,
+      padAngle,
+      colorScale,
+      style,
+    } = victoryStyles.pieChart;
+
+    const padding = {
+      top: height * padVertical,
+      bottom: height * padVertical,
+      right: width * padHorizontal,
+      left: width * padHorizontal,
+    };
+
+    const widthPadded = width * (1 - padHorizontal);
+
+    return (
+      <VictoryPie
+        width={width}
+        height={height}
+        padding={padding}
+        padAngle={padAngle}
+        innerRadius={widthPadded * innerRadius}
+        labelRadius={widthPadded * labelRadius}
+        colorScale={colorScale}
+        labelComponent={<VictoryLabel style={style} />}
+        data={report.data}
+      />
+    );
+  };
+
+  const renderVisualisation = () => {
+    if (report === null || !width || !height) return null;
+    switch (report.type) {
+      case 'PieChart':
+        return renderPieChart();
+      default:
+        return null;
+    }
+  };
+  return (
+    <View style={localStyles.ChartContainer} onLayout={onLayout}>
+      {renderVisualisation()}
```

Comment:
You haven't taken out most of this logic which was there when this component was switching between different charts.

There's basically no need for the functions `renderVisualisation` or `renderPieChart` - in fact `renderPieChart` is basically what this component should be.

So I suggest, take the code from `renderPieChart` out of the function, and place it inside the `PieChart` component.. and basically remove the rest! hopefully that makes sense

Reply:
I made those changes. Thanks btw ;-)